skfair/metrics/fairness_report.py

```diff
@@ -0,0 +1,57 @@
+from collections import defaultdict
+
+from sklearn.metrics import confusion_matrix
+import numpy as np
+
+def fairness_report(y_true, y_pred, groups, group_names=None):
+    unique_groups = set(groups)
+
+    grouped_data = defaultdict(list)
+    for i, (y_t, y_p) in enumerate(zip(y_true, y_pred)):
+        group_key = group_names[i] if group_names else groups[i]
+        grouped_data[group_key].append((y_t, y_p))
+
+    outputs = []
+    for group_name, group_data in grouped_data.items():
+        y_true_group, y_pred_group = zip(*group_data)
+
+        conf_matrix = confusion_matrix(y_true_group, y_pred_group, labels=[0,1])
+        tn, fp, fn, tp = conf_matrix.ravel()
+        eps = 1e-10
+        precision = tp / (tp + fp + eps)
+        recall = tp / (tp + fn + eps)
+
+        outputs.append({
```

Comment:
I'm curious if we want to allow the users to subset/add metrics that are of interest here. @MBrouns got thoughts on this? I don't want to overload the parameters but it feels like there may be some valid wiggle room here.

Reply:
that is a good point, any thoughts on a set of metrics that we want to allow?

Reply:
I may need to think about this. One attitude is that this is not set in stone so this is not a blocker. But looking at the [documentation on metrics](https://scikit-fairness.netlify.app/api/metrics.html) we currently support `p_percent_score` and `equal_opportunity`. It may be that you only care about one, or both. In the future I expect this list to grow too. So it feels like we should have some flexbility but at the same time I don't want to make it easy for people to over-rule a good standard. There's also the concern that you may want to keep functions simple.

Reply:
I think it is a good idea to customise the metrics as the metrics you might want to track vary on a case by case basis in fairness. Should we add a parameter metrics that accepts a list with string names for those metrics we support? Or should I just add these two metrics into the report for now?

Reply:
Why not a list that accepts functions? That way the user can also pass in their own custom function. Or a dictionary with names and functions?

Reply:
Sounds good, will implement